provider/pkg/provider/container.go

```diff
@@ -30,8 +32,275 @@ import (
+		if err != nil {
+			return fmt.Errorf("error sending request: %s: %q %+v", err, uri, body)
+		}
+
+		_, err = p.waitForResourceOpCompletion(urn, res.Update.CloudAPIOperation, op)
+		if err != nil {
+			return errors.Wrapf(err, "waiting for completion")
+		}
+		return nil
+	}
+}
+
+func updateClusterNestedField(diffPropPath, targetPropPath resource.PropertyPath,
+	defaultIfMissing propValueIfMissingFunc, operationSuffix, httpMethod string) containerCustomUpdateHandlerFunc {
+	return func(p *googleCloudProvider,
+		urn resource.URN,
+		label string,
+		res *resources.CloudAPIResource,
+		newInputs,
+		oldState resource.PropertyMap) error {
+		logging.V(9).Infof("[%s] updateNodePoolConfig called for property path: %q", label, diffPropPath.String())
+		tok := urn.Type()
+		moduleMember := tokens.ModuleMember(tok)
+		version := moduleMember.Name().String()
+		endpoint := resources.CloudAPIEndpoint{
+			Template: fmt.Sprintf(`https://container.googleapis.com/%s/`+
+				`projects/{projectsId}/locations/{locationsId}/clusters/{clustersId}%s`, version, operationSuffix),
+			Values: append(res.Create.Endpoint.Values,
+				resources.CloudAPIResourceParam{
+					Name:    "clusterId",
+					SdkName: "name",
+					Kind:    "path",
+				}),
+		}
+		uri, err := endpoint.URI(newInputs.Mappable(), oldState.Mappable())
+		if err != nil {
+			return err
+		}
+		// Get the new config object to copy the value for fieldName from
+		newConfigField, ok := diffPropPath.Get(resource.NewObjectProperty(newInputs))
+		if !ok {
+			logging.V(9).Infof("[%s] no update found at path: %q. Assuming the field is deleted.", label,
+				diffPropPath.String())
+			newConfigField = defaultIfMissing(oldState, diffPropPath)
+		}
+		// Overwrite the config object where the one value for fieldName has been updated.
+		newConfig := resource.NewObjectProperty(resource.NewPropertyMapFromMap(nil))
+		if targetPropPath == nil {
+			targetPropPath = diffPropPath
+		}
+		// Now updatedConfigField essentially is a config object with just one field - `fieldName`
+		updatedConfigField, ok := targetPropPath.Add(newConfig, newConfigField)
+		if !ok {
+			return fmt.Errorf("failed to inject config at property path: %q", targetPropPath.String())
+		}
+		logging.V(9).Infof("[%s] staged property value: %+v", label, updatedConfigField.ObjectValue().Mappable())
+		body := p.prepareAPIInputs(updatedConfigField.ObjectValue(), oldState, map[string]resources.CloudAPIProperty{
+			"config": {Flatten: true},
+		})
+		logging.V(9).Infof("[%s] cluster update request body: %v", label, body)
+		op, err := retryRequest(p.client, "PUT", uri, "", body)
+		if err != nil {
+			return fmt.Errorf("error sending request: %s: %q %+v", err, uri, body)
+		}
+
+		_, err = p.waitForResourceOpCompletion(urn, res.Update.CloudAPIOperation, op)
+		if err != nil {
+			return errors.Wrapf(err, "waiting for completion")
+		}
+		return nil
+	}
+}
+
+var clusterUpdateOrder = []string{
```

Comment:
How was this order determined? Is there a link we could include as a comment?

Reply:
Added a comment. Somewhat copied off the terraform provider but not strictly.